### PR TITLE
[NUI] remove StateValueCollection from Selector

### PR DESCRIPTION
**Pull request by dasg34**
State-Value pair is now added to Selector.StateValueList without
duplicate check.

You need to use `Selector.StateValueList` when custom state-value pair is added
to Selector. and it is now able to add custom state and pre-defined
state in the same initializer.

Before:
```
Selector<string> textSelector = new Selector<string>()
{
    Normal = "Defalut",
    { ControlState.Pressed, "Pressed!" }, // build error
    { ControlState.Focused, "Focused!" }  // build error
};
```

After:
```
Selector<string> textSelector = new Selector<string>()
{
    Normal = "Default!",
    StateValueList =
    {
        { ControlState.Pressed, "Pressed!" },
        { ControlState.Focused, "Focused!" }
    }
};
```

Also, this patch fixes a bunch of CA2227(Collection properties should be read
only) warnings.

### API Changes ###

 - ACR:

### CODE REVIEW

File: src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs [hunk 1]
```diff
@@ -27,10 +28,31 @@ namespace Tizen.NUI.BaseComponents
...
-    public class Selector<T> : StateValueCollection<T>
+    public class Selector<T>
     {
         private readonly bool cloneable = typeof(T).IsAssignableFrom(typeof(ICloneable));
 
+        /// <summary>
+        /// The list for adding state-value pair.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public IList<StateValuePair<T>> StateValueList { get; private set; } = new List<StateValuePair<T>>();
```
Comment (rabbitfor) [inline src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs:39]: Why don't we just use List instead of IList?

**Comment by rabbitfor** [comment]
Can you show me the xaml example about the custom control state in the selector?

File: src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs (hunk 1 above)
Comment (dasg34) [reply src/Tizen.NUI/src/public/BaseComponents/Style/Selector.cs:39]: MS design-guidelines not recommend `List<T>`.
```
❌ DO NOT use ArrayList or List<T> in public APIs.

These types are data structures designed to be used in internal implementation, not in public APIs. 
List<T> is optimized for performance and power at the cost of cleanness of the APIs and flexibility. 
For example, if you return List<T>, you will not ever be able to receive notifications when client code modifies the collection. 
Also, List<T> exposes many members, such as BinarySearch, that are not useful or applicable in many scenarios. 
The following two sections describe types (abstractions) designed specifically for use in public APIs.
```
(https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/guidelines-for-collections)

**Comment by dasg34** [comment]
> Can you show me the xaml example about the custom control state in the selector?

In xaml, custom control state does not work now. i will add control state converter later.

**Review by AdunFang — APPROVED** [review]
It is OK.

**Comment by dasg34** [comment]
> Can you show me the xaml example about the custom control state in the selector?

Here is the custom state xaml example.
https://github.com/Samsung/TizenFX/pull/2002

**Comment by woohyun** [comment]
@rabbitfor recommended to merge this commit a week later (after Sep 21).

**Comment by dongsug-song** [comment]
@dasg34 
Hello,
could you solve conflicts?

**Pull request merged**